daft/dataframe/dataframe.py

```diff
@@ -89,8 +83,11 @@ def _builder(self) -> LogicalPlanBuilder:
         if self._result_cache is None:
             return self.__builder
         else:
+            num_partitions = self._result_cache.num_partitions()
+            # Partition set should always be set on cache entry.
+            assert num_partitions is not None
```

Comment:
Nit: we can inline the comment into the assert

```suggestion
            assert num_partitions is not None, "Partition set should always be set on cache entry"
```